src/AddIns/Uno.UI.Lottie/ts/Uno.UI.Lottie.ts

```diff
@@ -132,42 +146,57 @@ namespace Uno.UI {
 		}
 
 		private static createAnimation(properties: LottieAnimationProperties): RunningLottieAnimation {
-			var existingAnimation = this._runningAnimations[properties.elementId];
+			const existingAnimation = Lottie._runningAnimations[properties.elementId];
 			if (existingAnimation) {
 				// destroy any previous animation
 				existingAnimation.animation.destroy();
 				existingAnimation.animation = null;
 			}
 
-			const config = this.getPlayerConfig(properties);
-			const animation = this._player.loadAnimation(config);
+			const config = Lottie.getPlayerConfig(properties);
+			const animation = Lottie._player.loadAnimation(config);
 
 			const runningAnimation = {
 				animation: animation,
 				properties: properties
 			};
 
-			this._runningAnimations[properties.elementId] = runningAnimation;
+			Lottie._runningAnimations[properties.elementId] = runningAnimation;
 
 			(animation as any).addEventListener("complete", (e: any) => {
 				Lottie.raiseState(animation);
 			});
 
-			if (animation.isLoaded) {
+			(animation as any).addEventListener("loopComplete", (e: any) => {
 				Lottie.raiseState(animation);
-			} else {
-				(animation as any).addEventListener("data_ready", (e: any) => {
-					Lottie.raiseState(animation);
-				});
-			}
+			});
+
+			(animation as any).addEventListener("segmentStart", (e: any) => {
+				Lottie.raiseState(animation);
+			});
+
+			(animation as any).addEventListener("data_ready", (e: any) => {
+				Lottie._numberOfFrames = animation.totalFrames;
+				Lottie.raiseState(animation);
+			});
+
+			Lottie.raiseState(animation);
 
 			return runningAnimation;
 		}
 
+		private static getStateString(animation: Lottie.AnimationItem): string {
+			const duration = animation.getDuration(false);
+
+			const state = `${animation.animationData.w}|${animation.animationData.h}|${animation.isLoaded}|${animation.isPaused}|${duration}`;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Exceeds maximum line length of 120](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5436651)